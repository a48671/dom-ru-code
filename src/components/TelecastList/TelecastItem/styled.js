import styled from 'styled-components';
import {colors} from '../../../variables/colors';

export const Wrapper = styled.div.attrs({className: 'telecast-item'})`
	display: flex;
	padding: 10px 20px;
	align-items: flex-start;
	border-bottom: 1px solid ${colors.gray};
	background-color: ${
		props => {
			if(props.type === 'isNow') return colors.blue;
			if(props.type === 'isOver') return colors.pink;
			else return 'initial';
		}
	};
`;

export const Time = styled.div.attrs({className: 'telecast-time'})`
	width: 70px;
	font-size: 22px;
`;

export const Title = styled.h1.attrs({className: 'head__title'})`
	width: calc(100% - 70px);
	font-size: 20px;
	font-weight: 600;
	color: ${colors.dark};
	padding-left: 30px;
	overflow: hidden;
`;