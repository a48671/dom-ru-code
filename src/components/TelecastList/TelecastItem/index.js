import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

import {Wrapper, Time, Title} from './styled'

class TelecastItem extends PureComponent {
	render() {

		const {title, time, type} = this.props;

		return (
			<Wrapper type={type}>
				<Time>
					{time}
				</Time>
				<Title>
					{title}
				</Title>
			</Wrapper>
		);
	}
}

TelecastItem.propTypes = {
	title: PropTypes.string,
	time: PropTypes.string,
	type: PropTypes.string
};

TelecastItem.defaultProps = {
	title: '',
	img: '',
	altImg: '',
	type: 'inFuture'
};

export default TelecastItem;