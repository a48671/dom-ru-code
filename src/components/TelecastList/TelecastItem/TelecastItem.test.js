import React from 'react';
import renderer from 'react-test-renderer';

import TelecastItem from './index';

describe('TelecastItem', () => {
  it('render correctly', () => {
    const TelecastItemComponent = renderer
      .create(<TelecastItem title="Title" time="11:00" type="isOver" />)
      .toJSON();
    expect(TelecastItemComponent).toMatchSnapshot();
  });
});
