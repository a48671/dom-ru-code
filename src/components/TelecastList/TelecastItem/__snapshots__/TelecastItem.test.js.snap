// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelecastItem render correctly 1`] = `
<div
  className="sc-bdVaJa telecast-item jefSHH"
  type="isOver"
>
  <div
    className="sc-bwzfXH telecast-time gAnrFN"
  >
    11:00
  </div>
  <h1
    className="sc-htpNat head__title kCVlzB"
  >
    Title
  </h1>
</div>
`;
